components/clp-py-utils/clp_py_utils/initialize-spider-db.py

```diff
@@ -0,0 +1,301 @@
+    CONSTRAINT `task_dep_parent` FOREIGN KEY (`parent`) REFERENCES `tasks` (`id`) ON UPDATE NO ACTION ON DELETE CASCADE,
+    CONSTRAINT `task_dep_child` FOREIGN KEY (`child`) REFERENCES `tasks` (`id`) ON UPDATE NO ACTION ON DELETE CASCADE
+);
+""",
+    """
+CREATE TABLE IF NOT EXISTS `task_instances`
+(
+    `id`         BINARY(16) NOT NULL,
+    `task_id`    BINARY(16) NOT NULL,
+    `start_time` TIMESTAMP  NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+    CONSTRAINT `instance_task_id` FOREIGN KEY (`task_id`) REFERENCES `tasks` (`id`) ON UPDATE NO ACTION ON DELETE CASCADE,
+    PRIMARY KEY (`id`)
+);
+""",
+    """
+CREATE TABLE IF NOT EXISTS `scheduler_leases`
+(
+    `scheduler_id` BINARY(16) NOT NULL,
+    `task_id`      BINARY(16) NOT NULL,
+    `lease_time`   TIMESTAMP  NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+    CONSTRAINT `lease_scheduler_id` FOREIGN KEY (`scheduler_id`) REFERENCES `schedulers` (`id`) ON UPDATE NO ACTION ON DELETE CASCADE,
+    CONSTRAINT `lease_task_id` FOREIGN KEY (`task_id`) REFERENCES `tasks` (`id`) ON UPDATE NO ACTION ON DELETE CASCADE,
+    INDEX (`scheduler_id`),
+    PRIMARY KEY (`scheduler_id`, `task_id`)
+);
+""",
+    """
+CREATE TABLE IF NOT EXISTS `data_locality`
+(
+    `id`      BINARY(16)  NOT NULL,
+    `address` VARCHAR(40) NOT NULL,
+    KEY (`id`) USING BTREE,
+    CONSTRAINT `locality_data_id` FOREIGN KEY (`id`) REFERENCES `data` (`id`) ON UPDATE NO ACTION ON DELETE CASCADE
+);
+""",
+    """
+CREATE TABLE IF NOT EXISTS `data_ref_driver`
+(
+    `id`        BINARY(16) NOT NULL,
+    `driver_id` BINARY(16) NOT NULL,
+    KEY (`id`) USING BTREE,
+    KEY (`driver_id`) USING BTREE,
+    CONSTRAINT `data_driver_ref_id` FOREIGN KEY (`id`) REFERENCES `data` (`id`) ON UPDATE NO ACTION ON DELETE CASCADE,
+    CONSTRAINT `data_ref_driver_id` FOREIGN KEY (`driver_id`) REFERENCES `drivers` (`id`) ON UPDATE NO ACTION ON DELETE CASCADE
+);
+""",
+    """
+CREATE TABLE IF NOT EXISTS `data_ref_task`
+(
+    `id`      BINARY(16) NOT NULL,
+    `task_id` BINARY(16) NOT NULL,
+    KEY (`id`) USING BTREE,
+    KEY (`task_id`) USING BTREE,
+    CONSTRAINT `data_task_ref_id` FOREIGN KEY (`id`) REFERENCES `data` (`id`) ON UPDATE NO ACTION ON DELETE CASCADE,
+    CONSTRAINT `data_ref_task_id` FOREIGN KEY (`task_id`) REFERENCES `tasks` (`id`) ON UPDATE NO ACTION ON DELETE CASCADE
+);
+""",
+    """
+CREATE TABLE IF NOT EXISTS `client_kv_data`
+(
+    `kv_key`    VARCHAR(64)    NOT NULL,
+    `value`     BLOB(60000) NOT NULL,
+    `client_id` BINARY(16)     NOT NULL,
+    PRIMARY KEY (`client_id`, `kv_key`)
+);
+""",
+    """
+CREATE TABLE IF NOT EXISTS `task_kv_data`
+(
+    `kv_key`  VARCHAR(64)    NOT NULL,
+    `value`   BLOB(60000) NOT NULL,
+    `task_id` BINARY(16)     NOT NULL,
+    PRIMARY KEY (`task_id`, `kv_key`),
+    CONSTRAINT `kv_data_task_id` FOREIGN KEY (`task_id`) REFERENCES `tasks` (`id`) ON UPDATE NO ACTION ON DELETE CASCADE
+);
+""",
+]
+
+
+def main(argv):
```

Comment:
can we split this into helpers to improve readability?

something like:
```
def _load_clp_config(config_path: pathlib.Path):
    pass

def _validate_database_credentials(clp_config: ClpConfig) -> bool:
    pass

def _setup_spider_database(clp_config: ClpConfig) -> bool:
    pass
```

it's true we're repeating certain logics in multiple places than this scripts - we can also leave it as-is in this PR, submit an issue and fix all together a future PR